### PR TITLE
clients: upload from rsemgr to uploadclient - #3084

### PR DESCRIPTION
**Pull request by TomasJavurek**
clients: upload from rsemgr to uploadclient - #3084

### CODE REVIEW

**Comment by bari12** [comment]
@TWAtGH please have a look as well.

File: lib/rucio/client/uploadclient.py [hunk 1]
```diff
@@ -289,9 +300,9 @@ def upload(self, items, summary_file_path=None, traces_copy_out=None):
                 json.dump(final_summary, summary_file, sort_keys=True, indent=1)
 
         if num_succeeded == 0:
-            raise NoFilesUploaded()
+            raise NoFilesUploaded(str(collections.Counter(summary_exception)))
```
Comment (TWAtGH) [inline lib/rucio/client/uploadclient.py:321]: Could we omit these changes for now and think again about how we pass these error information back?

**Review by TWAtGH — CHANGES_REQUESTED** [review]

File: lib/rucio/client/uploadclient.py (hunk 1 above)
Comment (TomasJavurek) [reply lib/rucio/client/uploadclient.py:321]: Well. I did already two versions. We greed on previous one, we agreed on this one. I am not going to apply another already third one.

Comment (TWAtGH) [reply lib/rucio/client/uploadclient.py:321]: I didn't agree ^^ Back then I started doing my own version based on your commit but I wasn't able to finish it and now I also don't have time to finish it.

**Comment by TomasJavurek** [comment]
Since Exceptions are using kwargs, I thought I could do just easily:
 raise NotAllFilesUploaded(summary=summary_exception)
and the summary becomes attribute of the exception. But I got:
TypeError: must be type, not classobj

**Comment by bari12** [comment]
Closing this since it has been resubmitted

**Pull request closed without merging**